hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java

```diff
@@ -205,12 +204,12 @@ public static Protocol getHttpProtocol(Event event) {
     String fullUrl = EnrichedSpanUtils.getFullHttpUrl(event).orElse(null);
     if (fullUrl != null) {
       try {
-        URI uri = new URI(fullUrl);
-        Protocol protocol = NAME_TO_PROTOCOL_MAP.get(uri.getScheme().toUpperCase());
+        URL url = new URL(fullUrl);
```

Comment:
Do we have protocol flowing in as a first class field in the event? If yes, let's refactor the function to get the protocol from the first class field and just use `NAME_TO_PROTOCOL_MAP` to get the `Protocol` value

Reply:
Yes, we have the protocol as a first class field. Made this change.